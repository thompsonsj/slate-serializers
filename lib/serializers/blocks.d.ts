export declare const isBlock: (tagName: string) => boolean;
