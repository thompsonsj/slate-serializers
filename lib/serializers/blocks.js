"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isBlock = void 0;
// See: <https://html.spec.whatwg.org/#the-css-user-agent-style-sheet-and-presentational-hints>
// From: https://www.npmjs.com/package/rehype-minify-whitespace
const blocks = [
    'address',
    'article',
    'aside',
    'blockquote',
    'body',
    'br',
    'caption',
    'center',
    'col',
    'colgroup',
    'dd',
    'dialog',
    'dir',
    'div',
    'dl',
    'dt',
    'figcaption',
    'figure',
    'footer',
    'form',
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
    'head',
    'header',
    'hgroup',
    'hr',
    'html',
    'legend',
    'li',
    'li',
    'listing',
    'main',
    'menu',
    'nav',
    'ol',
    'optgroup',
    'option',
    'p',
    'plaintext',
    'pre',
    'section',
    'summary',
    'table',
    'tbody',
    'td',
    'td',
    'tfoot',
    'th',
    'th',
    'thead',
    'tr',
    'ul',
    'wbr',
    'xmp', // Flow content, legacy
];
const isBlock = (tagName) => {
    return blocks.includes(tagName);
};
exports.isBlock = isBlock;
