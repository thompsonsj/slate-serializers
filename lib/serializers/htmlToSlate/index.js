"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.htmlToSlate = void 0;
const slate_hyperscript_1 = require("slate-hyperscript");
const htmlparser2_1 = require("htmlparser2");
const domhandler_1 = require("domhandler");
const domutils_1 = require("domutils");
const css_select_1 = require("css-select");
const default_1 = require("../../config/htmlToSlate/default");
const domhandler_2 = require("../../utilities/domhandler");
const utilities_1 = require("../../utilities");
const blocks_1 = require("../blocks");
const whitespace_1 = require("./whitespace");
const deserialize = ({ el, config = default_1.config, index = 0, childrenLength = 0, context = '', }) => {
    var _a, _b;
    if (el.type !== htmlparser2_1.ElementType.Tag && el.type !== htmlparser2_1.ElementType.Text) {
        return null;
    }
    const currentEl = el;
    const nodeName = (0, domutils_1.getName)(currentEl);
    const childrenContext = (0, whitespace_1.getContext)(nodeName) || context;
    const isLastChild = index === childrenLength - 1;
    const isWithinTextNodes = ((_a = currentEl.prev) === null || _a === void 0 ? void 0 : _a.type) === htmlparser2_1.ElementType.Text && ((_b = currentEl.next) === null || _b === void 0 ? void 0 : _b.type) === htmlparser2_1.ElementType.Text;
    if (nodeName === 'br' && config.convertBrToLineBreak && context !== 'preserve') {
        return [(0, slate_hyperscript_1.jsx)('text', { text: context ? '\n' : '' }, [])];
    }
    const children = currentEl.childNodes
        ? currentEl.childNodes
            .map((node, i) => deserialize({
            el: node,
            config,
            index: i,
            childrenLength: currentEl.childNodes.length,
            context: childrenContext,
        }))
            .filter((element) => element)
            .filter((element) => !isSlateDeadEnd(element))
            .map((element) => addTextNodeToEmptyChildren(element))
            .flat()
        : [];
    if ((0, domutils_1.getName)(currentEl) === 'body') {
        return (0, slate_hyperscript_1.jsx)('fragment', {}, children);
    }
    if (config.elementTags[nodeName]) {
        const attrs = config.elementTags[nodeName](currentEl);
        // elementAttributeMap is a convenient config for making changes to all elements
        const style = (0, utilities_1.getNested)(config, 'elementStyleMap');
        if (style) {
            Object.keys(style).forEach((slateKey) => {
                const cssProperty = style[slateKey];
                const cssValue = (0, domhandler_2.extractCssFromStyle)(currentEl, cssProperty);
                if (cssValue) {
                    attrs[slateKey] = cssValue;
                }
            });
        }
        return (0, slate_hyperscript_1.jsx)('element', attrs, children);
    }
    if (config.textTags[nodeName] || el.type === htmlparser2_1.ElementType.Text) {
        const attrs = gatherTextMarkAttributes({ el: currentEl });
        const text = (0, whitespace_1.processTextValue)({
            text: (0, domutils_1.textContent)(el),
            context: childrenContext,
            isInlineStart: index === 0,
            isInlineEnd: Number.isInteger(childrenLength) && isLastChild,
            isNextSiblingBlock: (el.next && (0, domhandler_1.isTag)(el.next) && (0, blocks_1.isBlock)(el.next.tagName)) || false,
        });
        if (text === '') {
            return null;
        }
        if ((0, whitespace_1.isAllWhitespace)(text)) {
            if (config.filterWhitespaceNodes && !childrenContext) {
                return null;
            }
            if (config.convertBrToLineBreak) {
                if (currentEl.prev && (0, domutils_1.getName)(currentEl.prev) === 'br') {
                    return null;
                }
                if (currentEl.next && (0, domutils_1.getName)(currentEl.next) === 'br') {
                    return null;
                }
            }
        }
        return [(0, slate_hyperscript_1.jsx)('text', { ...attrs, text }, [])];
    }
    return children;
};
const gatherTextMarkAttributes = ({ el, config = default_1.config }) => {
    let allAttrs = {};
    const children = (0, domutils_1.getChildren)(el);
    if (children.length) {
        ;
        [el, ...children.flat()].forEach((child) => {
            const name = (0, domutils_1.getName)(child);
            const attrs = config.textTags[name] ? config.textTags[name](child) : {};
            allAttrs = {
                ...allAttrs,
                ...attrs,
            };
        });
        if (children.length === 1 && (0, domutils_1.getChildren)(children[0]).length) {
            allAttrs = {
                ...allAttrs,
                ...gatherTextMarkAttributes({ el: children[0], config }),
            };
        }
    }
    else {
        const name = (0, domutils_1.getName)(el);
        const attrs = config.textTags[name] ? config.textTags[name](el) : {};
        allAttrs = {
            ...attrs,
        };
    }
    return allAttrs;
};
const htmlToSlate = (html, config = default_1.config) => {
    let slateContent = [];
    const handler = new domhandler_1.DomHandler((error, dom) => {
        if (error) {
            // Handle error
        }
        else {
            const updaters = config.htmlUpdaterMap ? Object.entries(config.htmlUpdaterMap) : [];
            for (const [selector, updater] of updaters) {
                (0, css_select_1.selectAll)(selector, dom).forEach((element) => {
                    if ((0, domhandler_1.isTag)(element)) {
                        const updated = updater(element);
                        if (updated !== null && updated !== element) {
                            if (typeof updated === 'string') {
                                (0, domutils_1.replaceElement)(element, (0, htmlparser2_1.parseDocument)(updated));
                            }
                            else {
                                (0, domutils_1.replaceElement)(element, updated);
                            }
                        }
                    }
                });
            }
            slateContent = dom
                .map((node) => deserialize({ el: node, config })) // run the deserializer
                .filter((element) => element) // filter out null elements
                .map((element) => {
                // ensure all top level elements have a children property
                if (!element.children) {
                    return {
                        children: element,
                    };
                }
                return element;
            })
                .filter((element) => !isSlateDeadEnd(element))
                .map((element) => addTextNodeToEmptyChildren(element));
        }
    });
    const parser = new htmlparser2_1.Parser(handler, { decodeEntities: true });
    let updatedHtml = html;
    if (config.htmlPreProcessString instanceof Function) {
        updatedHtml = config.htmlPreProcessString(html);
    }
    parser.write(updatedHtml);
    parser.end();
    return slateContent;
};
exports.htmlToSlate = htmlToSlate;
const isSlateDeadEnd = (element) => {
    const keys = Object.keys(element);
    if (!('children' in element))
        return false;
    return element.children.length === 0 && keys.length === 1;
};
const addTextNodeToEmptyChildren = (element) => {
    const keys = Object.keys(element);
    if (!('children' in element))
        return element;
    if (element.children.length === 0) {
        element.children.push({ text: '' });
    }
    return element;
};
