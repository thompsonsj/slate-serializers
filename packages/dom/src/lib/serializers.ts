import { AnyNode, Document } from 'domhandler'

import { config as defaultConfig } from './config/default'
import { Config } from './config/types'
import { convertSlate } from './utilities/convert-slate'

type SlateToDom = (node: any[], config?: Config) => AnyNode | ArrayLike<AnyNode>

export const slateToDom: SlateToDom = (node: any[], config = defaultConfig) => {
  if (!Array.isArray(node)) {
    return new Document([])
  }
  const document = node.map((n, index) =>
    convertSlate({
      node: n,
      config,
      isLastNodeInDocument: index === node.length - 1,
    }),
  )
  return document
}
