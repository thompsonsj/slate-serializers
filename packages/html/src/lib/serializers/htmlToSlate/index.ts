import { jsx } from 'slate-hyperscript'
import { parseDocument, Parser, ElementType } from 'htmlparser2'
import { ChildNode, DomHandler, Element, isTag, Node } from 'domhandler'
import { getChildren, getName, replaceElement, textContent } from 'domutils'
import { selectAll } from 'css-select'
import { Descendant } from 'slate'

import { Config } from './config/types'
import { config as defaultConfig } from './config/default'

import { extractCssFromStyle } from '@slate-serializers/dom'
import { getNested, isBlock } from '@slate-serializers/utilities'

import { Context, getContext, isAllWhitespace, processTextValue } from './whitespace'

interface Ideserialize {
  el: ChildNode
  config?: Config
  index?: number
  childrenLength?: number
  context?: Context
}

const deserialize = ({
  el,
  config = defaultConfig,
  index = 0,
  childrenLength = 0,
  context = '',
}: Ideserialize): any => {
  if (el.type !== ElementType.Tag && el.type !== ElementType.Text) {
    return null
  }

  const currentEl = el as Element
  const nodeName = getName(currentEl)
  const childrenContext = getContext(nodeName) || context

  const isLastChild = index === childrenLength - 1
  const isWithinTextNodes = currentEl.prev?.type === ElementType.Text && currentEl.next?.type === ElementType.Text

  if (nodeName === 'br' && config.convertBrToLineBreak && context !== 'preserve') {
    return [jsx('text', { text: context ? '\n' : '' }, [])]
  }

  const children = currentEl.childNodes
    ? currentEl.childNodes
        .map((node, i) =>
          deserialize({
            el: node,
            config,
            index: i,
            childrenLength: currentEl.childNodes.length,
            context: childrenContext,
          }),
        )
        .filter((element) => element)
        .filter((element) => !isSlateDeadEnd(element))
        .map((element) => addTextNodeToEmptyChildren(element))
        .flat()
    : []

  if (getName(currentEl) === 'body') {
    return jsx('fragment', {}, children)
  }

  if (config.elementTags[nodeName]) {
    const attrs: any = config.elementTags[nodeName](currentEl)
    // elementAttributeMap is a convenient config for making changes to all elements
    const style = getNested(config, 'elementStyleMap')
    if (style) {
      Object.keys(style).forEach((slateKey) => {
        const cssProperty = style[slateKey]
        const cssValue = extractCssFromStyle(currentEl, cssProperty)
        if (cssValue) {
          attrs[slateKey] = cssValue
        }
      })
    }
    return jsx('element', attrs, children)
  }

  if (config.textTags[nodeName] || el.type === ElementType.Text) {
    const attrs = gatherTextMarkAttributes({ el: currentEl,config })
    const text = processTextValue({
      text: textContent(el as Element),
      context: childrenContext as Context,
      isInlineStart: index === 0,
      isInlineEnd: Number.isInteger(childrenLength) && isLastChild,
      isNextSiblingBlock: (el.next && isTag(el.next) && isBlock(el.next.tagName)) || false,
    })
    if (text === '') {
      return null
    }
    if (isAllWhitespace(text)) {
      if (config.filterWhitespaceNodes && !childrenContext) {
        return null
      }
      if (config.convertBrToLineBreak) {
        if (currentEl.prev && getName(currentEl.prev as Element) === 'br') {
          return null
        }
        if (currentEl.next && getName(currentEl.next as Element) === 'br') {
          return null
        }
      }
    }
    return [jsx('text', { ...attrs, text }, [])]
  }

  return children
}

interface IgatherTextMarkAttributes {
  el: Element | ChildNode
  config?: Config
}

const gatherTextMarkAttributes = ({ el, config = defaultConfig }: IgatherTextMarkAttributes) => {
  let allAttrs = {}
  const children = getChildren(el)
  if (children.length) {
    ;[el, ...children.flat()].forEach((child) => {
      const name = getName(child as Element)
      const attrs = config.textTags[name] ? config.textTags[name](child as Element) : {}
      allAttrs = {
        ...allAttrs,
        ...attrs,
      }
    })
    if (children.length === 1 && getChildren(children[0]).length) {
      allAttrs = {
        ...allAttrs,
        ...gatherTextMarkAttributes({ el: children[0], config }),
      }
    }
  } else {
    const name = getName(el as Element)
    const attrs = config.textTags[name] ? config.textTags[name](el as Element) : {}
    allAttrs = {
      ...attrs,
    }
  }
  return allAttrs
}

export const htmlToSlate = (html: string, config: Config = defaultConfig) => {
  let slateContent: Descendant[] = []
  const handler = new DomHandler((error, dom) => {
    if (error) {
      // Handle error
    } else {
      const updaters = config.htmlUpdaterMap ? Object.entries(config.htmlUpdaterMap) : []
      for (const [selector, updater] of updaters) {
        selectAll<Node, Node>(selector, dom as any).forEach((element) => {
          if (isTag(element)) {
            const updated = updater(element)
            if (updated !== null && updated !== element) {
              if (typeof updated === 'string') {
                replaceElement(element, parseDocument(updated) as any)
              } else {
                replaceElement(element, updated)
              }
            }
          }
        })
      }
      slateContent = dom
        .map((node) => deserialize({ el: node, config })) // run the deserializer
        .filter((element) => element) // filter out null elements
        .map((element) => {
          // ensure all top level elements have a children property
          if (!element.children) {
            return {
              children: element,
            }
          }
          return element
        })
        .filter((element) => !isSlateDeadEnd(element))
        .map((element) => addTextNodeToEmptyChildren(element))
    }
  })
  const parser = new Parser(handler, { decodeEntities: true })
  let updatedHtml = html
  if (config.htmlPreProcessString instanceof Function) {
    updatedHtml = config.htmlPreProcessString(html)
  }
  parser.write(updatedHtml)
  parser.end()
  return slateContent
}

const isSlateDeadEnd = (element: { children: [] }) => {
  const keys = Object.keys(element)
  if (!('children' in element)) return false
  return element.children.length === 0 && keys.length === 1
}

const addTextNodeToEmptyChildren = (element: { children: any[] }) => {
  const keys = Object.keys(element)
  if (!('children' in element)) return element
  if (element.children.length === 0) {
    element.children.push({ text: '' })
  }
  return element
}
