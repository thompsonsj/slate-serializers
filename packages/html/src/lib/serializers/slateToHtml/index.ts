import serializer from 'dom-serializer'

import { slateToDom, slateToDomConfig as defaultConfig, type SlateToDomConfig } from '@slate-serializers/dom'

type SlateToHtml = (node: any[], config?: SlateToDomConfig) => string

export const slateToHtml: SlateToHtml = (node: any[], config = defaultConfig) => {
  const document = slateToDom(node, config)
  return serializer(document, {
    encodeEntities: 'encodeEntities' in config ? config.encodeEntities : false,
  })
}
