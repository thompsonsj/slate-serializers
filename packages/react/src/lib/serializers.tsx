import React, { ReactElement, JSXElementConstructor } from 'react'
import { Element, isTag, Text } from 'domhandler'
import { getName, textContent } from 'domutils'
import { ulid } from 'ulid'

import { slateToDomConfig, type SlateToDomConfig, convertSlate } from '@slate-serializers/dom'
import { config as slateToReactConfig } from './config/default'
import type { Config as SlateToReactConfig } from './config/types'

interface ISlateToReact {
  node: any[]
  config?: SlateToDomConfig
  reactConfig?: SlateToReactConfig
}

export const SlateToReact = ({ node, config = slateToDomConfig, reactConfig = slateToReactConfig }: ISlateToReact) => {
  if (!Array.isArray(node)) {
    return <></>
  }
  const document = node.map((n, index) =>
    convertSlate({
      node: n,
      ...{
        config,
        elementTransforms: {},
      },
      isLastNodeInDocument: index === node.length - 1,
      customElementTransforms: reactConfig.elementTransforms,
      transformText: (text) => transformText(text),
      transformElement: (element) => {
        return domElementToReactElement(element)
      },
      wrapChildren: (children) => children,
    }),
  )
  return document as any
}

const transformText = (text: Text | Element) => {
  if (isTag(text as Element)) {
    return React.createElement(getName(text as Element), {}, textContent(text as Element))
  }
  return <>{textContent(text as Text)}</>
}

const domElementToReactElement = (element: Element): ReactElement<any, string | JSXElementConstructor<any>> => {
  return React.createElement(
    getName(element),
    {
      /* Ensure all React elements have a unique key - is there a better way to do this? */
      key: ulid(),
      /* Provide all attributes as props */
      ...element.attribs,
      /* Convert key names for JSX compatibility */
      ...(element.attribs?.class && { className: element.attribs?.class }),
    },
    element.children as any,
  )
}
