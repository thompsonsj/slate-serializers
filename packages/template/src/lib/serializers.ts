import { config as slateToTemplateConfig } from './config/default'
import type { Config as SlateToTemplateConfig } from './config/types'
import { slateToHtml } from '@slate-serializers/html'

export const slateToTemplate = (node: any[], config: SlateToTemplateConfig = slateToTemplateConfig) => {
  if (!Array.isArray(node)) {
    return 
  }
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  return node.map((child: any) => {
    const type = child.type
    if (type && config.customElementSerializers?.[type]) {
      return config.customElementSerializers?.[type]({ node })
    } else {
      return slateToHtml([child], config)
    }
  })
}
