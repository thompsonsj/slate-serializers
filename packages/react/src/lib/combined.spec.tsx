import React from 'react'
import renderer from 'react-test-renderer'
import { fixtures as elementFixtures } from './combined/fixtures/elementTags'
import { fixtures as textFixtures } from './combined/fixtures/textTags'
import { fixtures as combinedFixtures } from './combined/fixtures/combined'
import { SlateToReact } from './serializers'

describe('Slate JSON to React transforms', () => {
  describe('Element tags', () => {
    const fixtures = elementFixtures
    for (const fixture of fixtures) {
      it(`${fixture.name}`, () => {
        const tree = renderer.create(<SlateToReact node={fixture.slate} />).toJSON()
        expect(tree).toMatchSnapshot()
      })
    }
  })
  describe('Text tags', () => {
    const fixtures = textFixtures
    for (const fixture of fixtures) {
      it(`${fixture.name}`, () => {
        const tree = renderer.create(<SlateToReact node={fixture.slate} />).toJSON()
        expect(tree).toMatchSnapshot()
      })
    }
  })
  describe('Combined', () => {
    const fixtures = combinedFixtures
    for (const fixture of fixtures) {
      it(`${fixture.name}`, () => {
        const tree = renderer.create(<SlateToReact node={fixture.slateOriginal} />).toJSON()
        expect(tree).toMatchSnapshot()
      })
    }
  })
})
