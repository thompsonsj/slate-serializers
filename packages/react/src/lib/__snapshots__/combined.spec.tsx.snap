// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slate JSON to React transforms Combined larger document 1 1`] = `
[
  <h2>
    Heading 2
  </h2>,
  "A regular paragraph.",
  "A ",
  <strong>
    paragraph
  </strong>,
  " with ",
  <strong>
    various
  </strong>,
  " text ",
  <s>
    marks
  </s>,
  ".",
  <h3>
    Heading 3
  </h3>,
  <ul>
    <li>
      <strong>
        Item 1
      </strong>
      : From a list
    </li>
    <li>
      <strong>
        Item 2
      </strong>
      : From a list
    </li>
  </ul>,
  <a
    href="https://docs.slatejs.org/"
  >
    Find out more about Slate
  </a>,
  ".",
  <h4>
    Heading 4
  </h4>,
]
`;

exports[`Slate JSON to React transforms Combined links nested in an unordered list 1`] = `
<ul>
  <li>
    <a
      href="https://github.com/thompsonsj/slate-serializers"
      target="_blank"
    >
      slate-serializers | GitHub
    </a>
  </li>
  <li>
    <a
      href="https://github.com/thompsonsj/slate-serializers"
      target="_blank"
    >
      slate-serializers | GitHub
    </a>
  </li>
</ul>
`;

exports[`Slate JSON to React transforms Combined text with inline link 1`] = `
[
  "Some text before an inline link ",
  <a
    href="https://github.com/thompsonsj/slate-serializers"
    target="_blank"
  >
    slate-serializers | GitHub
  </a>,
  ".",
]
`;

exports[`Slate JSON to React transforms Element tags blockquote 1`] = `
<blockquote>
  Blockquote
</blockquote>
`;

exports[`Slate JSON to React transforms Element tags headings 1`] = `
[
  <h1>
    Heading 1
  </h1>,
  <h2>
    Heading 2
  </h2>,
  <h3>
    Heading 3
  </h3>,
  <h4>
    Heading 4
  </h4>,
  <h5>
    Heading 5
  </h5>,
  <h6>
    Heading 6
  </h6>,
]
`;

exports[`Slate JSON to React transforms Element tags link 1`] = `
<a
  href="https://github.com/thompsonsj/slate-serializers"
>
  Slate Serializers | GitHub
</a>
`;

exports[`Slate JSON to React transforms Element tags nested unordered list 1`] = `
<ul>
  <li>
    Item 1
    <ul>
      <li>
        Nested item 1
      </li>
      <li>
        Nested item 2
      </li>
    </ul>
  </li>
  <li>
    Item 2
  </li>
</ul>
`;

exports[`Slate JSON to React transforms Element tags ordered list 1`] = `
<ol>
  <li>
    Item 1
  </li>
  <li>
    Item 2
  </li>
</ol>
`;

exports[`Slate JSON to React transforms Element tags paragraph 1`] = `
<p>
  Paragraph 1
</p>
`;

exports[`Slate JSON to React transforms Element tags unordered list 1`] = `
<ul>
  <li>
    Item 1
  </li>
  <li>
    Item 2
  </li>
</ul>
`;

exports[`Slate JSON to React transforms Text tags idiomatic text 1`] = `
<i>
  Idiomatic Text
</i>
`;

exports[`Slate JSON to React transforms Text tags pre 1`] = `
<pre>
  Pre
</pre>
`;

exports[`Slate JSON to React transforms Text tags strikethrough 1`] = `
<s>
  Strikethrough
</s>
`;

exports[`Slate JSON to React transforms Text tags strong 1`] = `
<strong>
  Bold
</strong>
`;

exports[`Slate JSON to React transforms Text tags strong and idiomatic text 1`] = `
<strong>
  Strong and Idiomatic Text
</strong>
`;

exports[`Slate JSON to React transforms Text tags strong and unarticulated annotation 1`] = `
<strong>
  Strong and Unarticulated Annotation
</strong>
`;

exports[`Slate JSON to React transforms Text tags strong in paragraph 1`] = `
<p>
  <strong>
    Bold
  </strong>
</p>
`;

exports[`Slate JSON to React transforms Text tags strong mid-sentence 1`] = `
[
  "Some ",
  <strong>
    bold text
  </strong>,
  " in a sentence.",
]
`;

exports[`Slate JSON to React transforms Text tags unarticulated annotation 1`] = `
<u>
  Unarticulated Annotation
</u>
`;
